pandas/core/generic.py

```diff
@@ -5213,6 +5245,9 @@ def __finalize__(self, other, method=None, **kwargs):
 
         """
         if isinstance(other, NDFrame):
+            for name in other.attrs:
+                self.attrs[name] = other.attrs[name]
```

Comment:
self.attrs.update(other) ?

Reply:
That'd work for now, but we'll need a for loop in my other PR since we'll have per-attr logic to resolve these.